### DISCUSSION TITLE
im studying ethr-did usage, plz help

### DISCUSSION THREAD

**Discussion by joeg9soft** [decentralized-identity/veramo#818] [Q&A]
hi im developing did project and i accidentally found uport github
and im using ethr-did , registry , resolver.
and my question is how to use signed functions in registry solidity.
for example, changeOwnerSigned(address identity, uint8 sigV, bytes32 sigR, bytes32 sigS, address newOwner) , there is this function
how do i make message for sign? i tried web3.eth.accounts.sign, but i dont know message format for sign() parameter
and other question is, do i have to make new geth account for these solidity functions?
on uport guide, it's using createKeyPair function, and this function doesn't make geth account on my geth file folder
and web3 method doesn't work when not matched geth account and keypair address

**Comment by mirceanis** [comment]
We don't have support for calling the `doSomethingSigned()` functions in Veramo; we currently only use the `doSomething()` variants.

The message to be signed is described in the [ERC1056 repo readme](https://github.com/uport-project/ethr-did-registry), for each one of the `*Signed()` functions. If you have questions about those methods, you can continue the discussion in https://github.com/uport-project/ethr-did-registry/issues/48.

You can use the non-signed variants of those methods, with the same security guarantees. There is still a signature being checked, but it is the transaction signature, not the special message signature. You haven't described your use-case but it seems like those methods might be more suitable for you with geth accounts.
